app/controllers/sessions_controller.rb

```diff
@@ -0,0 +1,48 @@
+# frozen_string_literal: true
+
+class SessionsController < ApplicationController
+  def create
+    @user = User.find_by(email: session_params[:email])
+
+    if @user&.authenticate(session_params[:password])
+      login!
+      render json: {
+        logged_in?: true,
+        user: @user
+      }
+    else
+      render json: {
+        status: 401,
+        errors: ['No user found']
+      }
+    end
+  end
+
+  def is_logged_in?
+    if logged_in? && current_user
+      render json: {
+        logged_in: true,
+        user: current_user
+      }
+    else
+      render json: {
+        logged_in: false,
+        message: 'no such user'
+      }
+    end
+  end
+
+  def destroy
+    logout!
```

Comment:
Do you think functions like logout could be moved from application to be private functions within this controller?

Reply:
As we discussed, I moved the logout function to sessions controller